tests/test_web_tools.py

```
@@ -61,3 +72,135 @@ def respond(request: httpx.Request) -> httpx.Response:
+            (),
+            {
+                "url": url,
+                "connect_url": url,
+                "host_header": "public.example",
+                "sni_hostname": "public.example",
+            },
+        )(),
+    )
+
+    with httpx.Client(transport=httpx.MockTransport(respond)) as client:
+        with web_tools._stream_public_response(client, "https://public.example/start") as response:
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)